sftp_test.go

```diff
@@ -26,3 +26,34 @@ func TestErrFxCode(t *testing.T) {
 		assert.Equal(t, statusErr.FxCode(), tt.fx)
 	}
 }
+
+func TestSupportedExtensions(t *testing.T) {
+	for _, supportedExtension := range supportedSFTPExtensions {
+		assert.True(t, isSFTPExtensionSupported(supportedExtension))
+	}
+}
+
+func TestExtensions(t *testing.T) {
+	testSFTPExtensions := []SSHExtensionPair{
+		{"hardlink@openssh.com", "1"},
+	}
+	err := SetSFTPExtensions(testSFTPExtensions)
+	assert.Nil(t, err)
+	assert.Equal(t, testSFTPExtensions, sftpExtensions)
+	invalidSFTPExtensions := []SSHExtensionPair{
+		{"invalid@openssh.com", "1"},
```

Comment:
Use `example.com` instead of `openssh.com` for this invalid case, in order to give a strong guarantee that this will not name-collide with a possible future extension.